src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/Packaging/ManagedProjectSystemPackage.cs

```diff
@@ -75,10 +78,15 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
 
         protected override void Dispose(bool disposing)
         {
-            if (disposing && _projectSelectorCookie != VSConstants.VSCOOKIE_NIL)
+            if (disposing)
             {
-                _projectSelectorService?.UnregisterProjectSelector(_projectSelectorCookie);
-                _projectSelectorCookie = VSConstants.VSCOOKIE_NIL;
+                if (_projectSelectorCookie != VSConstants.VSCOOKIE_NIL)
+                {
+                    _projectSelectorService?.UnregisterProjectSelector(_projectSelectorCookie);
+                    _projectSelectorCookie = VSConstants.VSCOOKIE_NIL;
+                }
+
+                _solutionService?.Dispose();
```

Comment:
This is a MEF component, you don't own the lifetime of this, let MEF clean it up.

Reply:
👍